client/packages/attester/src/attester.ts

```diff
@@ -47,106 +58,232 @@ export class Attester {
 
     async listenEvents() {
         logger.info('Listening to events')
+
         // Subscribe to the NewAttestationMessageHash event
         this.circuit.sdk?.client.query.system.events(async (events) => {
             logger.debug({ events_count: events.length }, `Received events`)
             this.prometheus.eventsTotal.inc(events.length)
-            // Loop through the Vec<EventRecord>
-            await Promise.all(
-                events.map(async (record) => {
-                    // Extract the phase, event and the event types
-                    const { event } = record
-                    logger.debug(event.toHuman())
-
-                    if (event.section == 'attesters') {
-                        this.prometheus.eventsAttestationsTotal.inc().set({
-                            method: event.method,
-                        })
+
+            const comittee = await this.getCommittee()
+            this.checkIsInCommittee(comittee, this.keys.substrate.accountId)
+
+            const attesterEvents = events.filter(
+                async (event) => event.section == 'attesters'
+            )
+
+            this.prometheus.attestationsEvents.inc(attesterEvents.length)
+
+            await this.processAttestationEvents(attesterEvents)
+        })
+    }
+
+    private async processAttestationEvents(events: any) {
+        // Loop through the Vec<EventRecord>
+        await Promise.all(
+            events.map(async (record) => {
+                // Extract the phase, event and the event types
+                const { event } = record
+
+                switch (event.method) {
+                    case 'NewAttestationMessageHash': {
+                        const [targetId, messageHash, executionVendor] =
+                            event.data
+
+                        if (executionVendor.toString() == 'Substrate') {
+                            logger.warn('Substrate not implemented')
+                        } else if (executionVendor.toString() == 'Ed25519') {
```

Comment:
This execution vendor doesnt exist, only substrate and EVM ist available. Alternatively, you could use `GatewayVendor` which is chain specific (e.g Dot, Ksma, Eth, etc)

Reply:
Sounds good!
I will do it in next iteration.